### PR TITLE
NavigationSplitView

### PR DESCRIPTION
**Pull request by Amzd**
Currently `NavigationSplitView` does not support any `.navigationDestination` api unlike the SwiftUI counterpart but it introduces the basic split view built on Gtk Paned.

This PR also adds `Spacer`

### CODE REVIEW

File: Sources/Gtk/Widgets/Paned.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+open class Paned: Widget, Orientable {
+    public init(orientation: Orientation) {
+        super.init()
+        widgetPointer = gtk_paned_new(orientation.toGtkOrientation())
+    }
+
+    public var startChild: Widget? {
+        didSet {
+            gtk_paned_set_start_child(opaquePointer, startChild?.castedPointer())
```
Comment (Amzd) [inline Sources/Gtk/Widgets/Paned.swift:12]: hmm do we need to unset the parent of the previous child widget?

Comment (stackotter) [reply Sources/Gtk/Widgets/Paned.swift:12]: > hmm do we need to unset the parent of the previous child widget?

I think so, to keep the Swift side of the Gtk view hierarchy up to date with the C side

File: Examples/Split/SplitNavigationApp.swift [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+class NavigationAppState: Observable {
+    @Observed var selectedArea: SubjectArea?
+    @Observed var selectedDetail: Any?
+    @Observed var columns: Columns = .two
+}
+
+@main
+struct NavigationApp: App {
```
Comment (stackotter) [inline Examples/Split/SplitNavigationApp.swift]: Should be renamed to `SplitNavigationApp`

File: Package.swift [hunk 3]
```diff
@@ -77,6 +77,10 @@ let package = Package(
             name: "NavigationExample",
             targets: ["NavigationExample"]
         ),
+        .executable(
+            name: "SplitExample",
```
Comment (stackotter) [inline Package.swift:81]: Either rename this to match `SplitNavigationApp` or the other way around (I think `SplitExample` is suitable, so renaming to `SplitApp` would be fine).

File: Sources/SwiftCrossUI/Views/NavigationSplitView.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public init(@ViewContentBuilder sidebar: () -> SideBar,
+                @ViewContentBuilder detail: () -> Detail) {
+        body = ViewContent3(sidebar(), EmptyViewContent(), detail())
+    }
+}
+
+extension NavigationSplitView {
+    public init(@ViewContentBuilder sidebar: () -> SideBar,
+                @ViewContentBuilder content: () -> MiddleBar,
+                @ViewContentBuilder detail: () -> Detail) {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/Views/NavigationSplitView.swift]: Wrap arguments like this instead:
```swift
public init(
    @ViewContentBuilder sidebar: () -> SideBar,
    @ViewContentBuilder content: () -> MiddleBar,
    @ViewContentBuilder detail: () -> Detail
) {
```

File: Sources/SwiftCrossUI/Views/NavigationSplitView.swift [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            next.shrinkEndChild = false
+            next.position = 0
+            next.expandVertically = true
+
+            widget.endChild = next
+        }
+
+        widget.shrinkStartChild = false
+        widget.shrinkEndChild = false
+        widget.position = 0 // sets
```
Comment (stackotter) [inline Sources/SwiftCrossUI/Views/NavigationSplitView.swift]: what does the comment mean?

**Review by stackotter — CHANGES_REQUESTED** [review]
Looks good so far! This is going to be very useful. Just a few requested changes

File: Sources/SwiftCrossUI/Views/NavigationSplitView.swift (hunk 5 above)
Comment (Amzd) [reply Sources/SwiftCrossUI/Views/NavigationSplitView.swift]: updated to 
```swift
// Set the position to the farthest left possible.
// TODO: Allow setting the default offset (SwiftUI api: `navigationSplitViewColumnWidth(min:ideal:max:)`).
//       This needs frame modifier to be fledged out first
```

**Review by stackotter — APPROVED** [review]
Looks good to me 👍

**Pull request merged**